src/EndpointManagerStandalone.sol

```diff
@@ -4,55 +4,68 @@ pragma solidity >=0.6.12 <0.9.0;
-        // whether this endpoint is enabled
-        bool enabled;
-        uint8 index;
+    struct _Threshold {
+        uint8 num;
     }
 
-    // @dev Information about registered endpoints.
-    // This is the source of truth, we define a couple of derived fields below
-    // for efficiency.
-    mapping(address => EndpointInfo) public endpointInfos;
+    /// =============== STORAGE ===============================================
 
-    // @dev List of enabled endpoints.
-    // invariant: forall (a: address), endpointInfos[a].enabled <=> a in enabledEndpoints
-    address[] _enabledEndpoints;
+    bytes32 public constant THRESHOLD_SLOT = bytes32(uint256(keccak256("ntt.threshold")) - 1);
+
+    function _getThresholdStorage() private pure returns (_Threshold storage $) {
+        uint256 slot = uint256(THRESHOLD_SLOT);
+        assembly ("memory-safe") {
+            $.slot := slot
+        }
+    }
 
-    // invariant: forall (i: uint8), enabledEndpointBitmap & i == 1 <=> endpointInfos[i].enabled
-    uint64 _enabledEndpointBitmap;
+    /// =============== GETTERS/SETTERS ========================================
 
-    uint8 constant _MAX_ENDPOINTS = 64;
+    function setThreshold(uint8 threshold) external onlyOwner {
+        _Threshold storage _threshold = _getThresholdStorage();
+        _threshold.num = threshold;
+        _checkThresholdInvariants();
+    }
 
-    // @dev Total number of registered endpoints. This number can only increase.
-    // invariant: numRegisteredEndpoints <= MAX_ENDPOINTS
-    // invariant: forall (i: uint8),
-    //   i < numRegisteredEndpoints <=> exists (a: address), endpointInfos[a].index == i
-    uint8 _numRegisteredEndpoints;
+    /// @notice Returns the number of Endpoints that must attest to a msgId for
+    ///         it to be considered valid and acted upon.
+    function getThreshold() public view returns (uint8) {
+        _Threshold storage _threshold = _getThresholdStorage();
+        return _threshold.num;
+    }
 
-    // =========================================================================
+    function setEndpoint(address endpoint) external onlyOwner {
+        _setEndpoint(endpoint);
 
-    modifier onlyEndpoint() {
-        if (!endpointInfos[msg.sender].enabled) {
-            revert CallerNotEndpoint(msg.sender);
-        }
-        _;
+        _Threshold storage _threshold = _getThresholdStorage();
+        // We increase the threshold here. This might not be what the user
+        // wants, in which case they can call setThreshold() afterwards.
+        // However, this is the most sensible default behaviour, since
+        // this makes the system more secure in the event that the user forgets
+        // to call setThreshold().
+        _threshold.num += 1;
     }
 
-    constructor(address token, Mode mode, uint16 chainId) EndpointManager(token, mode, chainId) {
-        _checkEndpointsInvariants();
+    function removeEndpoint(address endpoint) external onlyOwner {
```

Comment:
Do we want to prevent removing an endpoint if 0 endpoints remain after the action? Not sure if that's something we should care about, seems like an admin responsibility.

Reply:
Imo it's okay -- say someone wants to fully replace their endpoints. They might want to reduce to 0, then re-add them.

I think we want to minimize cases where someone could shoot themselves in the foot and brick the contract, but 0 endpoints should be okay and easily fixable.

Reply:
this is actually a good point, because if the endpoints are all removed, then the threshold becomes 0, and any message is immediately accepted. Agree with @nik-suri that we should allow removing endpoints first then re-introducing them.
But we don't want to open up a time period where any message is accepted, so I added check https://github.com/wormhole-foundation/example-native-token-transfers/pull/9/commits/3b07332f6b2638c1e24735f929296917c57e9423 to prevent this

Reply:
nice catch @kcsongor !